Sources/NIOPosix/VsockAddress.swift

```diff
@@ -0,0 +1,289 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+#if canImport(Darwin)
+import CNIODarwin
+#elseif os(Linux)
+#if canImport(Glibc)
+import Glibc
+#elseif canImport(Musl)
+import Musl
+#endif
+import CNIOLinux
+#endif
+fileprivate let vsockUnimplemented = "VSOCK support is not implemented for this platform"
+
+// MARK: - Public API that's available on all platforms.
+
+/// A vsock socket address.
+///
+/// A socket address is defined as a combination of a Context Identifier (CID) and a port number.
+/// The CID identifies the source or destination, which is either a virtual machine or the host.
+/// The port number differentiates between multiple services running on a single machine.
+///
+/// For well-known CID values and port numbers, see `VsockAddress.ContextID` and `VsockAddress.Port` respectively.
```

Comment:
I think for docc you want:

```suggestion
/// For well-known CID values and port numbers, see ``VsockAddress/ContextID`` and ``VsockAddress/Port`` respectively.
```

Reply:
That's interesting and sounds a lot like a bug...

Reply:
It's not a bug, you need a proper reference. Docc IDs are case insensitive, so you need to disambiguate between `VsockAddress.Port` and `VsockAddress.port`. If you let Xcode autocomplete type this for you, it'll put the right thing in.

Reply:
> you need to disambiguate between VsockAddress.Port and VsockAddress.port. If you let Xcode autocomplete type this for you, it'll put the right thing in.

Well, Xcode autocomplete puts in <code>\`\`VsockAddress/Port\`\`</code> and resolves to the correct thing (jump to definition works fine).

Is there some other disambiguation I can use?

---

Here's the evidence that this doesn't work in CI (which I already tried).

Works fine currently:

```console
❯ docker-compose -f ./docker/docker-compose.yaml -f ./docker/docker-compose.2204.58.yaml run documentation-check
...
Generating documentation for 'NIOPosix'...
Building for debugging...
Build complete! (0.28s)
...

❯ echo $?
0
```

Then with this proposed change (even with Xcode autocompleting):

```console
❯ git diff
diff --git a/Sources/NIOPosix/VsockAddress.swift b/Sources/NIOPosix/VsockAddress.swift
index add18ffe3..442f938b6 100644
--- a/Sources/NIOPosix/VsockAddress.swift
+++ b/Sources/NIOPosix/VsockAddress.swift
@@ -33,7 +33,7 @@ fileprivate let vsockUnimplemented = "VSOCK support is not implemented for this
 /// The CID identifies the source or destination, which is either a virtual machine or the host.
 /// The port number differentiates between multiple services running on a single machine.
 ///
-/// For well-known CID values and port numbers, see `VsockAddress.ContextID` and `VsockAddress.Port` respectively.
+/// For well-known CID values and port numbers, see ``VsockAddress/ContextID`` and ``VsockAddress/Port`` respectively.
 public struct VsockAddress: Hashable, Sendable {
     /// The context ID associated with the address.
     public var cid: ContextID

❯ docker-compose -f ./docker/docker-compose.yaml -f ./docker/docker-compose.2204.58.yaml run documentation-check
...
Generating documentation for 'NIOPosix'...
Building for debugging...
Build complete! (0.27s)
Converting documentation...
/code/Sources/NIOPosix/VsockAddress.swift:36:84: error: Topic reference 'VsockAddress/Port' couldn't be resolved. No local documentation matches this reference.
Error: An error was encountered while compiling documentation
error: 'docc convert' invocation failed with a nonzero exit code: '1'

❯ echo $?
1
```

Reply:
Update: found the syntax, which appears to be <code>\`\`VsockAddress/Port-swift.struct\`\`</code>